docs/reference/docs/reindex.asciidoc

```diff
@@ -478,6 +478,10 @@ which defaults to a maximum size of 100 MB.
 `sort`:::
 (Optional, list) A comma-separated list of `<field>:<direction>` pairs to sort by before indexing.
 Use in conjunction with `max_docs` to control what documents are reindexed.
+deprecated[7.6, sort in reindex is deprecated] *NOTE: Sorting in reindex is deprecated.*
+Sorting in reindex was never guaranteed to index
+documents in order and prevents future evolution of reindex such as resilience and performance
+improvements. If used in combination with `max_docs`, consider using a query filter instead.
```

Comment:
I feel like the `deprecated` macro and the bold note are a bit redundant.

I'd place this content into a block `deprecated` macro like below.

```
`sort`:::
+
--
(Optional, list) A comma-separated list of `<field>:<direction>` pairs to sort by before indexing.
Use in conjunction with `max_docs` to control what documents are reindexed.

deprecated::[7.6,Sort in reindex is deprecated. Sorting in reindex was never guaranteed to index documents in order and prevents reindex such as resilience and performance improvements. If used in combination with `max_docs`, consider using a query filter instead.]
--
```

Reply:
I tried this now, but I get an error saying:

```
element listitem: validity error : Element listitem content does not follow the DTD
```
which is why I originally went for the simpler solution. My asciidoc skills are not super-advanced, so maybe there is a trick to make this work?

Reply:
I was able to get this working on my end without that error.

<img width="725" alt="Screen Shot 2019-11-27 at 3 43 43 PM" src="https://user-images.githubusercontent.com/40268737/69758153-be782700-112c-11ea-86f3-e5d2c636036b.png">

Here's the commit on my fork if that helps:
https://github.com/jrodewig/elasticsearch/commit/7111829c9d08cba24af5dc776c53e958a7490a58

Feel free to cherry-pick it or copy/paste the contents.

There are a few tricks:
* The delimiter setup is needed.
* The `deprecated` text must be on a single line.

Reply:
What you have is workable though. If you're not able to get the streamlined deprecated note working, I wouldn't hold up this PR. We can reformat it as a separate issue.

Reply:
Thanks so much @jrodewig , worked out fine on one line (had to also escape the comma), see: aa2a7e04354b4389ac886ee42f1ecbf634d5fb8a